scene/resources/particles_material.cpp

```diff
@@ -378,14 +378,18 @@ void ParticlesMaterial::_update_shader() {
 	code += "	CUSTOM.x = base_angle * degree_to_rad;\n"; // angle
 	code += "	CUSTOM.y = 0.0;\n"; // phase
 	code += "	CUSTOM.w = (1.0 - lifetime_randomness * rand_from_seed(alt_seed));\n";
-	code += "	CUSTOM.z = (tex_anim_offset) * mix(anim_offset_min, anim_offset_max, anim_offset_rand);\n"; // animation offset (0-1)
+	code += "	CUSTOM.z = (tex_anim_offset) * mix(anim_offset_min, anim_offset_max, anim_offset_rand);\n\n"; // animation offset (0-1)
+
+	code += "   if (RESTART_ROT_SCALE) {\n";
+	code += "		TRANSFORM = mat4(vec4(1,0,0,0),vec4(0,1,0,0),vec4(0,0,1,0),vec4(0,0,0,1));\n";
+	code += "   }\n\n";
```

Comment:
Some style issues in the generated code (should use tabs for indentation, and spaces after commas, though that one was a pre-existing issue)):
```suggestion
	code += "	if (RESTART_ROT_SCALE) {\n";
	code += "		TRANSFORM = mat4(vec4(1, 0, 0, 0), vec4(0, 1, 0, 0), vec4(0, 0, 1, 0), vec4(0, 0, 0, 1));\n";
	code += "	}\n\n";
```

That being said... Isn't `mat4()` an identity matrix in GLSL?

Reply:
Bump.

Reply:
I changed the spaces to tabs and changed the way the default 1.0 scale is applied to the particle.

Reply:
Did you try `mat4(1.0)` for the identity matrix? https://thebookofshaders.com/glossary/?search=mat4